File: frontend/src/api/official.ts
import { apiRequest } from './client'
import type { CreateGameResponse } from './types'

export type OfficialGame = {
  slug: string
  title: string
  description: string
  play_url: string
  thumbnail_url: string | null
}

export const officialApi = {
  /** GET /official-games */
  async list(locale?: string): Promise<OfficialGame[]> {
    const q = locale ? `?locale=${encodeURIComponent(locale)}` : ''
    return apiRequest<OfficialGame[]>(`/official-games${q}`)
  },

  /** POST /games/fork/{slug} — 404 时抛错由 UI toast */
  fork(slug: string, accessToken: string) {
    return apiRequest<CreateGameResponse>(`/games/fork/${encodeURIComponent(slug)}`, {
      method: 'POST',
      token: accessToken,
      body: {},
    })
  },
}

File: backend/app/schemas/official.py
from pydantic import BaseModel


class OfficialGameItem(BaseModel):
    """OfficialGameItem 列表项 DTO。

    场景：分页/列表 API 的单条记录结构。"""

    slug: str
    title: str
    description: str
    play_url: str
    thumbnail_url: str | None = None
